client/extensions/woocommerce/components/custom-tree-selector/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import CustomTreeNode from './custom-tree-node';
+
+function renderPlaceholders() {
+	return (
+		<div>
+			<div className="custom-tree-selector__placeholder" />
+			<div className="custom-tree-selector__placeholder" />
+			<div className="custom-tree-selector__placeholder" />
+		</div>
+	);
+}
+
+const CustomTreeSelector = ( {
+	nodes,
+	onNodeSelect,
+} ) => {
+	const renderedNodes = nodes && nodes.map( ( node ) => {
+		return (
+			<CustomTreeNode
+				key={ node.key }
+				node={ node }
+				onNodeSelect={ onNodeSelect }
+			/>
+		);
+	} );
+
+	return (
+		<div className="custom-tree-selector">
+			{ ! nodes && renderPlaceholders() }
+			{ renderedNodes }
+		</div>
+	);
+};
+
+CustomTreeSelector.PropTypes = {
+	nodes: PropTypes.array,
```

Comment:
Should this be `isRequired`?

Reply:
Actually this is intentional. If `nodes` is falsy, placeholders are displayed by design. Do you think that's okay? It seems to work well with redux selectors fairly well that way.

Reply:
It's not really what I'd expect - I could pass in an empty array while waiting for data to load, which would satisfy `PropTypes.array.isRequired` and still be falsy to display placeholders.

(In testing this I accidentally used `node` instead of `nodes`, and it just showed me placeholders. Having the prop be required would've pointed that mistake out in the console)